training/MNISTFashionMicroservice/src/server/training.py

```diff
@@ -2,20 +2,68 @@
 import tempfile
 import shutil
 import requests
+import sys
+import logging
+import json
 
 from src.server.dependency import ModelData
 import tensorflow as tf
 
+class StreamToLogger(object):
+    """
+    Fake file-like stream object that redirects writes to a logger instance.
+    Source: https://stackoverflow.com/a/39215961
+    """
+    def __init__(self, logger, level):
+       self.logger = logger
+       self.level = level
+       self.linebuf = ''
+
+    def write(self, buf):
+       for line in buf.rstrip().splitlines():
+          self.logger.log(self.level, line.rstrip())
+
+    def flush(self):
+        pass
 
 def train_model(training_id, model_data: ModelData):
-    acc = [-1]
-    val_acc = [-1]
 
-    loss = [-1]
-    val_loss = [-1]
-    print("Save:" + str(model_data.save))
+    """
+    Train model(s) based on a given model and hyperparameters
+    Now supporting two hyperparameters which are 
+    - Optimizer and learning_rate
+    """
+
+    # SET LOGGER TO PRINT TO STDOUT AND WRITE TO FILE
+    logging.basicConfig(
+        level=logging.DEBUG,
+        format="%(asctime)s [%(levelname)s] %(message)s",
+        handlers=[
+            logging.FileHandler("/log/{}.log".format(training_id)),
+            logging.StreamHandler(sys.stdout)
+        ]
+    )
+    log = logging.getLogger('foobar')
```

Comment:
What is this `foobar` logger? Is there a better name to give it?

Reply:
i think this a placeholder logger for when we implement one. that can be a separate MR

Reply:
> What is this `foobar` logger? Is there a better name to give it?

Sure, I will change the name.   


> i think this a placeholder logger for when we implement one. that can be a separate MR

Thanks, what does MR mean?

Reply:
Pushed. The logger name is changed.